File: StreamDeck_Tests/StreamDeck_Python/data_publisher.py
import time
import sys
import rti.connextdds as dds
from data import statistic_data

class statistic_dataPublisher:

    @staticmethod
    def run_publisher(domain_id: int, sample_count: int):

        participant = dds.DomainParticipant(domain_id)

        topic = dds.Topic(participant, "statistic_data", statistic_data)

        writer = dds.DataWriter(participant.implicit_publisher, topic)

        sample = statistic_data()

        for count in range(sample_count):
            # Catch control-C interrupt
            try:
                # Modify the data to be sent here
                sample.auto_flag = count
                
                print(f"Writing statistic_data, count {count}")
                writer.write(sample)
                time.sleep(1)
            except KeyboardInterrupt:
                break

        print("preparing to shut down...")


if __name__ == "__main__":
    statistic_dataPublisher.run_publisher(
            domain_id=0,
            sample_count=sys.maxsize)

File: StreamDeck_Tests/Vehicle_CPP/data_publisher.cxx
#include <iostream>
#include <stdlib.h>

#include <dds/pub/ddspub.hpp>
#include <rti/util/util.hpp>      
#include <rti/config/Logger.hpp> 

#include "application.hpp"  
#include "data.hpp"

void run_publisher_application(){
    
    int domain_id = 0;

    dds::domain::DomainParticipant participant(domain_id);

    dds::pub::Publisher publisher(participant);

    dds::topic::Topic< ::statistic_data> topic(participant, "statistic_data");

    dds::pub::DataWriter< ::statistic_data> writer(publisher, topic);

    ::statistic_data data;
    
    int count = 0;

    while (!application::shutdown_requested) {

        count += 1;
        int auto_flag;

        if (count % 3 == 1) {
            auto_flag = 0;
        }
        else {
            auto_flag = 1;
        }

        srand(time(nullptr));

        data.height(rand()%10);
        data.depth(rand()%10);
        data.auto_flag(auto_flag);

        std::cout << "Writing ::statistic_data" << std::endl;

        writer.write(data);

        rti::util::sleep(dds::core::Duration(1));
    }
}

int main(int argc, char *argv[]){

    try {
        run_publisher_application();
    } catch (const std::exception& ex) {
        std::cerr << "Exception in run_publisher_application(): " << ex.what()
        << std::endl;
        return EXIT_FAILURE;
    }

    
    dds::domain::DomainParticipant::finalize_participant_factory();

    return EXIT_SUCCESS;
}
